packages/transaction-controller/src/utils/utils.ts

```diff
@@ -191,3 +191,24 @@ export function normalizeGasFeeValues(
     maxPriorityFeePerGas: normalize(gasFeeValues.maxPriorityFeePerGas),
   };
 }
+
+/**
+ * Picks the middle element from an array of gas fees.
+ *
+ * @param fees - An array of gas fees.
+ * @returns The middle element from the array.
+ */
+export function pickMiddleFeeElement(
```

Comment:
I'd vote to keep this specific to the smart gas fee flow as it has no other usages plus is loosely coupled to the smart transaction fee type.

Not sure if we'll need this now though as we just want to pick the highest estimate and set that to `medium`.

Reply:
Done in https://github.com/MetaMask/core/pull/4009/commits/c57cfd91bc45e7ef7e02329d100193a21032fa86